### PR TITLE
add mapi.3igames.mail.ru to subdoc exceptions

### PR DESCRIPTION
**Pull request by mrchnk**
3igames.mail.ru is the publishing platform for online games (flash mostly). That platform is also used as a bridge for a game to other sites (like vk.com, odnoklassniki.ru, my.mail.ru, and others). So document containing flash plugin belongs to another document on domain mapi.3igames.mail.ru that belongs to the final destination domain (3igames.mail.ru, vk.com, odnoklassniki.ru, etc).

And mapi.3igames.mail.ru blocked by wildcard mail.ru domain from list flashsubdoc.txt

Please, add mapi.3igames.mail.ru to flashsubdocexceptions.txt to allow online players play flash content in Firefox.

### CODE REVIEW

**Comment by cpeterso** [comment]
Thanks for the bug report, @mrchnk! We'll look into this subdomain. Can you share a direct link to a Flash game (that doesn't require an account) so we can reproduce the problem and confirm the subdoc exception fixes it?

**Comment by mrchnk** [comment]
I am afraid that every game on that publishing platform requires account on any of these: my.mail.ru, vk.com, odnoklassniki.ru, 3igames.mail.ru

Here are links to the same game
- https://games.mail.ru/app/3205
- https://my.mail.ru/apps/754367
- https://vk.com/app5909956
- https://ok.ru/game/arena-legend
- https://3igames.mail.ru/games/arl/play

**Comment by cpeterso** [comment]
@mrchnk, I spoke with the engineering team. If we add mapi.3igames.mail.ru to the subdoc exception whitelist, Flash content loaded from mapi.3igames.mail.ru will still require the user to manually click to activate it in the subdocument. Will your Flash games still be able to work correctly if they require manual user activation?

See [this SUMO KB article](https://support.mozilla.org/en-US/kb/why-do-i-have-click-activate-plugins) for more information about Firefox's click-to-activate user experience. If the Flash content's dimensions are large enough, Firefox's ["Run Adobe Flash" button](https://prod-cdn.sumo.mozilla.net/uploads/gallery/images/2018-01-12-15-51-22-564015.png) will be displayed in the page. If the Flash content is smaller than the button (e.g. a "Flash pixel" used to talk to a server but not show a user interface), Firefox will instead show a ["door hanger" prompt](https://prod-cdn.sumo.mozilla.net/uploads/gallery/images/2018-01-13-02-44-10-0ba3b2.png) that users might not see.

**Comment by mrchnk** [comment]
I guess that adding subdoc exception would be a better solution than just block flash content completely for that page. Seems like "Click to activate" becomes common for flash content in Firefox and other major browsers, and it would not be a problem for players.

**Comment by cpeterso** [comment]
@felipc, can you please review this PR before I ask @ckolos to merge it to Shavar? This is the subdoc exception we discussed in our Friday meeting.

**Review by felipc — APPROVED** [review]

**Comment by cpeterso** [comment]
Thanks, @felipc!

@ckolos, when you have a chance, can you please merge this PR? I still have merge permissions for this repo, but I think it would be better if you do the merge since you are more aware of any changes that are happening on the Shavar backend.

Tomorrow is a new Firefox release (58), but I don't see any reason to delay this Shavar update for that.

**Comment by cpeterso** [comment]
ckolos, thanks for merging!

@mrchnk, the blocklist update will probably take about 24 hours to reach users. Please let us know if this update fixes your Flash games.

**Comment by mrchnk** [comment]
Thanks everyone for the help. 
After clean installation of Firefox problem with our games on that platform disappeared. It is not blocked anymore.

**Comment by cpeterso** [comment]
I'm glad it works now. Thanks for reporting this issue!

**Pull request merged**
